scene-lib/src/scenelib/annotations/el/LocalLocation.java

```diff
@@ -1,91 +1,219 @@
 package scenelib.annotations.el;
 
+import org.objectweb.asm.Label;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
 import java.util.Objects;
 
+import org.checkerframework.checker.nullness.qual.Nullable;
+
 /**
  * A {@link LocalLocation} holds location information for a local
- * variable: slot index, scope start, and scope length.
+ * variable: slot index, scope start, and scope end.
  */
 public final class LocalLocation {
     /**
```

Comment:
I would add a comment that the representation is 3 arrays, rather than an array of triples, because that is what ASM uses.

Reply:
I'm confused - where do you see an array of triples in ASM?  I see the three arrays, for example, in MethodVisitor: 
public AnnotationVisitor visitLocalVariableAnnotation(int typeRef,
                                                      TypePath typePath,
                                                      Label[] start,
                                                      Label[] end,
                                                      int[] index,
                                                      java.lang.String descriptor,
                                                      boolean visible)

Reply:
Sorry for ambiguity.  I meant "the representation is not an array of triples, but is 3 arrays because that is what ASM uses."